### PR TITLE
Add coverage subdirectory

### PR DESCRIPTION
**Pull request by j-rivero**
This PR adds a `coverage/` directory to host code related to coverage build for packages in the quality level list. README.md include information about how to use and explanation about the different files.

Related to https://github.com/ros2/ros2_documentation/pull/737

### CODE REVIEW

**Comment by j-rivero** [comment]
The `rosinstall_generator` command does not include the sros2 dependency present as `test_depend` in `test_security`. Adding it manually to the rosinstall file by now.

File: coverage/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## Files
+
+ * `quality_level.txt`:
+   package names inside the qualitly level list
+ * `coverage_package_list`:
+   manual selection of packages from ros2.repos that depend on quality level packages
+   to be included as part of coverage builds
+ * `coverage_generated.rosinstall`:
+   roinstall file generated by generate_rosinstall_for_coverage.bash
```
Comment (dirk-thomas) [inline coverage/README.md:23]: :-1: from me to commit the `coverage_generated.rosinstall` file which contains redundant information from the rosdistro.

Comment (nuclearsandwich) [reply coverage/README.md:23]: I believe the intent is to commit it so that it can be used in CI until CI can run the necessary rosinstall_generator command itself.

Comment (dirk-thomas) [reply coverage/README.md:23]: Why can't CI not run the necessary `rosinstall_generator` command atm?

Comment (nuclearsandwich) [reply coverage/README.md:23]: I don't see any code in ros2/ci that calls rosinstall_generator. Are you proposing we add a new feature to ros2/ci which does so as part of the coverage work?

Comment (dirk-thomas) [reply coverage/README.md:23]: Yes.

Comment (dirk-thomas) [reply coverage/README.md:23]: ~~Yes.~~

Let me rephrase this: it isn't clear yet how this should be used - which my above question for more context was about.

**Comment by j-rivero** [comment]
Closed in favor of https://github.com/ros2/ci/pull/466
